ui/src/widgets/ui-markdown/UIMarkdown.vue

```diff
@@ -1,51 +1,126 @@
-<template>
-    <vue-markdown class="nrdb-ui-widget nrdb-ui-markdown" :source="props.content" />
-</template>
-
 <script>
-import VueMarkdown from 'vue-markdown-render'
-import { mapState } from 'vuex'
+import DOMPurify from 'dompurify'
+import hljs from 'highlight.js'
+import 'highlight.js/styles/default.css'
+import { marked } from 'marked'
+import { markedHighlight } from 'marked-highlight'
+
+import { h } from 'vue'
+import { mapGetters, mapState } from 'vuex' // eslint-disable-line import/order
 
 import { useDataTracker } from '../data-tracker.js'
 
+marked.use(markedHighlight({
+    langPrefix: 'hljs language-',
+    highlight (code, lang) {
+        const language = hljs.getLanguage(lang) ? lang : 'plaintext'
+        return hljs.highlight(code, { language }).value
+    }
+}))
+
 export default {
     name: 'DBUIMarkdown',
-    components: {
-        VueMarkdown
-    },
     inject: ['$socket'],
     props: {
         id: { type: String, required: true },
         props: { type: Object, default: () => ({}) }
     },
+    setup (props) {
+        useDataTracker(props.id)
+        return () => h({
+            props: ['id', 'props'],
+            errorCaptured: (err, vm, info) => {
+                console.error('errorCaptured', err, vm, info)
+                return false
+            },
+            template: `<div class="nrdb-ui-markdown">${DOMPurify.sanitize(marked.parse(props.props.content))}</div>`,
```

Comment:
I expected to be able to do `{{ msg?.payload || 'Placeholder' }}` but that unfortunately makes the widget crash. 

Is that a limitation of the approach here, or _should_ this work?

Reply:
The error is occurring at the `marked.parse` step where the output becomes:
```
<td>{{ msg?.payload</td>
```

Reply:
Dont know why marked would be doing that to the `{{ }}` 

I know VUE doesnt like optional chaining `xxx?.yyy` in certain places but what you are saying is that `marked` is messing the format first anyhow?  Odd.  

An this does not happen when using `{{ msg?.payload }}` right? (only happens when you introduce a default value via the `||` shortcut?) 

PS: I made sure `msg` was always an object (see the computed prop) so there is no real need for `msg?.` now - but it is still something that needs to be understood as a user might well expect to be able to enter `{{ msg?.payload || 'Placeholder' }}`

Reply:
It was actually the `||` - it was treating it as a table and getting very confused. Just pushed a fix to support `||`. It's a little hardcoded, but okay for now.